### PR TITLE
feat(flame_behaviors)!: migrate to flame v1.10.0

### LINKED ISSUE #29: refactor: convert input behaviors to the new APIs once they hit stable.
**Description**

The input behaviors, like `DraggableBehavior` and `TappableBehavior` are using the old input system. There is now a (experimental) new API for handing drag and tap events: 
- [drag events](https://docs.flame-engine.org/1.3.0/flame/inputs/drag-events.html)
- [tap events](https://docs.flame-engine.org/1.3.0/flame/inputs/tap-events.html)

Ones these hit as stable we should refactor these behaviors to work with the new API. This will introduce a breaking change in `flame_behaviors`.

**Requirements**

- [ ] Refactor input behaviors to the new API
- [ ] Test their functionality
- [ ] Write migration documentation

### PR DESCRIPTION
**Pull request by wolfenrain**
## Description

Updated to flame v1.10.0

Closes #29 

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [x] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by erickzanardo — DISMISSED** [review]

File: packages/flame_behaviors/example/pubspec.yaml [hunk 1]
```diff
@@ -4,10 +4,10 @@ version: 1.0.0+1
 publish_to: none
 
 environment:
-  sdk: ">=2.17.0 <3.0.0"
+  sdk: ">=3.0.0 <4.0.0"
 
 dependencies:
-  flame: ^1.5.0
+  flame: ^1.8.0
```
Comment (luanpotter) [inline packages/flame_behaviors/example/pubspec.yaml:10]: why not use 1.10 here too?

**Review by luanpotter — APPROVED** [review]
LGTM

Can't say I agree with the `if` style changes but that is just personal preference :P

File: packages/flame_behaviors/example/pubspec.yaml (hunk 1 above)
Comment (wolfenrain) [reply packages/flame_behaviors/example/pubspec.yaml:10]: Originally started as v1.8.0 upgrade, will change that!

**Comment by wolfenrain** [comment]
> Can't say I agree with the `if` style changes but that is just personal preference :P

That wasn't supposed to happen, so I reverted them!

**Review by erickzanardo — APPROVED** [review]

**Review by renancaraujo — APPROVED** [review]

**Pull request merged**